### DISCUSSION TITLE
Optimized css production build with nextjs ssg

### DISCUSSION THREAD

**Discussion by cipriancaba** [tailwindlabs/tailwindcss#9527] [Help]
I am building an ecomm site with nextjs and tailwindcss fed with data from a headless cms. Tailwindcss is using the jit compiler to generate just the necessary css, however considering the dynamic nature of our site generation, we don't have access to all that information before the build step.

However, we do statically generate most of our pages so it would be ideal if we could run the css build step of the nextjs build pipeline after the html files were generated. 

Is there a simple way to just point tailwindcss to the generated nextjs build and simply generate the css after the initial build step? I'm sure someone must have hit this issue before and has a solution for it

TLDR:
- All our codebase classes get properly whitelisted and generated
- We have some tailwind classes in the cms that are not getting whitelisted due to the fact that they are not available until after the build process when nextjs generates the static files

**Comment by wongjn** [comment]
You could look at running a Tailwind build using the Tailwind CLI after your initial build. Add `content` globs that point to your compiled Next.js files to your Tailwind config so that it sufficiently covers the built files.
